src/d/actor/d_a_wall.cpp

```diff
@@ -4,99 +4,354 @@
+    Vec cullMax = m_cull_size[mType * 2 + 1];
+    fopAcM_setCullSizeBox(this, cullMin.x, cullMin.y, cullMin.z, cullMax.x, cullMax.y, cullMax.z);
+    fopAcM_setCullSizeFar(this, 1.0f);
+    mStts.Init(0xFF, 0xFF, this);
+    set_tri();
+    dComIfG_Bgsp()->Regist(mpBgW, this);
+    set_mtx();
+    mpBgW->Move();
+    mSwitchIndex = fopAcM_GetParam(this) & 0xFF;
+    mState = false;
 }
 
-/* 00000380-000004EC       .text _create__8daWall_cFv */
 s32 daWall_c::_create() {
-    /* Nonmatching */
+    fopAcM_SetupActor(this, daWall_c);
+    mType = fopAcM_GetParam(this) >> 8;
+    mSwitchIndex = fopAcM_GetParam(this) & 0xFF;
+    bool isSwitch = dComIfGs_isSwitch(mSwitchIndex, fopAcM_GetHomeRoomNo(this));
+    if (isSwitch || mSwitchIndex == 0xff) {
+        return cPhs_ERROR_e;
+    }
+
+    s32 phase_state = dComIfG_resLoad(&mPhs, m_arcname[mType]);
+
+    if (phase_state == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, m_heapsize[mType])) {
+            return cPhs_ERROR_e;
+        }
+        CreateInit();
+    }
+
+    return phase_state;
 }
 
 /* 00000800-00000880       .text set_mtx__8daWall_cFv */
 void daWall_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(mScale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
 }
 
 /* 00000880-000008F0       .text setMoveBGMtx__8daWall_cFv */
 void daWall_c::setMoveBGMtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(fopAcM_GetShapeAngle_p(this)->y);
+    mDoMtx_stack_c::scaleM(mScale);
+    mDoMtx_copy(mDoMtx_stack_c::get(), mMtx);
 }
 
 /* 000008F0-00000980       .text _execute__8daWall_cFv */
 BOOL daWall_c::_execute() {
     /* Nonmatching */
+    typedef void (daWall_c::*procFunc)();
+    static procFunc mode_proc[] = {
+        &daWall_c::mode_wait,
+        &daWall_c::mode_break,
+    };
+
+    (this->*mode_proc[mState])();
+    return 1;
 }
 
 /* 00000980-00000A58       .text mode_wait__8daWall_cFv */
 void daWall_c::mode_wait() {
-    /* Nonmatching */
+    for (int i = 0; i < 2; i++) {
+        if (mTri[i].ChkTgHit()) {
+            cCcD_Obj * at = mTri[i].GetTgHitObj();
+            if (at != NULL && at->ChkAtType(AT_TYPE_BOMB)) {
+                set_effect();
+                set_se();
+                break;
+            } else {
+                mTri[i].ClrTgHit();
+            }
+        }
+    }
+
+    for (int i = 0; i < 2; i++) {
+        dComIfG_Ccsp()->Set(&mTri[i]);
+    }
 }
 
 /* 00000A58-00000B14       .text mode_break__8daWall_cFv */
 void daWall_c::mode_break() {
-    /* Nonmatching */
+    if (mBreakCounter != 0) {
+        mBreakCounter = mBreakCounter + 1;
+        switch (mType) {
+        case 0:
+        case 1:
+        case 2:
+            int tmp;
+            if ((mBreakCounter > 10) && (tmp = cLib_chaseF(&mpDst, 0.0, 2.222222), tmp != 0)) {
```

Comment:
```suggestion
            if ((mBreakCounter > 10) && cLib_chaseF(&mDst, 0.0f, 2.222222f)) {
```
tmp variable shouldn't be necessary I think, unless this fails to match for some reason?
also these are singles not doubles